app/controllers/security_group_controller.rb

```diff
@@ -107,11 +107,8 @@ def create_finished
 
   def delete_security_groups
     assert_privileges("security_group_delete")
-
-    security_groups = if @lastaction == "show_list" || (@lastaction == "show" && @layout != "security_group")
-                        find_checked_records_with_rbac(SecurityGroup)
-                      else
-                        [find_record_with_rbac(SecurityGroup, params[:id])]
+    security_groups = if @lastaction == "show_list" || (@lastaction == "show" && @layout != "security_group") [find_records_with_rbac(SecurityGroup, checked_or_params)]
```

Comment:
@stanas `[find_records_with_rbac(SecurityGroup, checked_or_params)]` is wrong. This would result having an array inside of array (the `find_records_with_rbac` method return an array with records).